apps/web/app/api/v1/client/displays/route.ts

```diff
@@ -1,88 +1,61 @@
-/*
-THIS FILE IS WORK IN PROGRESS
-PLEASE DO NOT USE IT YET
-*/
-
 import { responses } from "@/lib/api/response";
-import { prisma } from "@formbricks/database";
+import { InvalidInputError } from "@formbricks/errors";
+import { createDisplay } from "@formbricks/lib/services/displays";
 import { capturePosthogEvent } from "@formbricks/lib/posthogServer";
+import { getSurvey } from "@formbricks/lib/services/survey";
+import { getTeamDetails } from "@formbricks/lib/services/teamDetails";
 import { NextResponse } from "next/server";
+import { TDisplay, TDisplayInput, ZDisplayInput } from "@formbricks/types/v1/displays";
+import { transformErrorToDetails } from "@/lib/api/validator";
 
 export async function OPTIONS(): Promise<NextResponse> {
   return responses.successResponse({}, true);
 }
 
 export async function POST(request: Request): Promise<NextResponse> {
-  const { surveyId, personId, environmentId } = await request.json();
 
-  if (!surveyId) {
-    return responses.missingFieldResponse("surveyId", true);
-  }
+  const displayInput: TDisplayInput = await request.json();
+  const inputValidation = ZDisplayInput.safeParse(displayInput);
 
-  if (!environmentId) {
-    return responses.missingFieldResponse("environmentId", true);
+  if (!inputValidation.success) {
+    return responses.badRequestResponse(
+      "Fields are missing or incorrectly formatted",
+      transformErrorToDetails(inputValidation.error),
+      true
+    );
   }
 
-  // get teamId from environment
-  const environment = await prisma.environment.findUnique({
-    where: {
-      id: environmentId,
-    },
-    select: {
-      product: {
-        select: {
-          team: {
-            select: {
-              id: true,
-              memberships: {
-                select: {
-                  userId: true,
-                  role: true,
-                },
-              },
-            },
-          },
-        },
-      },
-    },
-  });
+  // find environmentId from surveyId
+  let survey;
 
-  if (!environment) {
-    return responses.notFoundResponse("Environment", environmentId, true);
+  try {
+    survey = await getSurvey(displayInput.surveyId);
```

Comment:
when you use `unknown` type at the top, you need to use `inputValidation.data.surveyId` here to have it typesafe. you can also change the naming at the top to have it consistent:

```
const jsonInput: unknown = await request.json();
const inputValidation = ZDisplayInput.safeParse(jsonInput);
// ... (validation)
const displayInput = inputValidation.data;
```